aws/resource_aws_ssm_patch_group.go

```diff
@@ -59,7 +59,7 @@ func resourceAwsSsmPatchGroupRead(d *schema.ResourceData, meta interface{}) erro
 
 	found := false
 	for _, t := range resp.Mappings {
-		if *t.PatchGroup == d.Id() {
+		if fmt.Sprintf("%s:%s", *t.PatchGroup, *t.BaselineIdentity.BaselineId) == d.Id() {
```

Comment:
I would be in favour of having a function generate the ID. Then the function can be called in L46 too. This will help in the future if we want to change the format of the ID, the change will only need to be made in one place.

Reply:
This doesn't appear to be how it's handled elsewhere in the codebase, so I went for consistency. 

Some examples:

`
aws/resource_aws_guardduty_ipset.go:    d.SetId(fmt.Sprintf("%s:%s", detectorID, *resp.IpSetId))
aws/resource_aws_glue_catalog_database.go:      d.SetId(fmt.Sprintf("%s:%s", catalogID, name))
aws/resource_aws_iam_user_policy.go:    d.SetId(fmt.Sprintf("%s:%s", *request.UserName, *request.PolicyName))
aws/resource_aws_network_interface_sg_attachment.go:            d.SetId(fmt.Sprintf("%s_%s", sgID, interfaceID))
aws/resource_aws_servicequotas_service_quota.go:        d.SetId(fmt.Sprintf("%s/%s", serviceCode, quotaCode))
aws/resource_aws_ses_identity_policy.go:        d.SetId(fmt.Sprintf("%s|%s", identity, policyName))
aws/resource_aws_route53_zone_association.go:   d.SetId(fmt.Sprintf("%s:%s", *req.HostedZoneId, *req.VPC.VPCId))
`

All of the examples I looked at did have a function to decode/parse the state ID back into its component parts, but that appeared to be only due to using those parts in a query. 

Unfortunately, although the DescribePatchGroups operation allows a filter, I have yet to find a Key that actually works, so with no way to filter the query, a function to decode the state ID seems pointless.

Reply:
No problem - if that's what is done elsewhere I'm happy with the consistency 👍